hazelcast/include/hazelcast/client/serialization/pimpl/compact/compact_impl.h

```diff
@@ -574,6 +572,46 @@ struct schema_of
 template<typename T>
 const schema schema_of<T>::schema_v = schema_of<T>::build_schema();
 
+template<typename T>
+schema
+build_schema(const T& object)
+{
+    schema_writer schema_writer(hz_serializer<T>::type_name());
+    serialization::compact_writer writer =
+      create_compact_writer(&schema_writer);
+    serialization::hz_serializer<T>::write(object, writer);
+    return std::move(schema_writer).build();
+}
+
+template<typename T>
+class class_to_schema
+{
+public:
+    static std::atomic<bool> is_initialized;
```

Comment:
Can you make them private and change set method to `void set(object)` where you build the schema if needed?

Reply:
Fixed.